File: aidp-java-client-2x/src/main/java/com/oracle/aidataplatform/dp/model/PromptToolConfiguration.java
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.

package com.oracle.aidataplatform.dp.model;


/**
 * Tool configurations are set by the agent developer when they create the tool. | The agent does not see those configurations and can not modify their values
**/
@javax.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=PromptToolConfiguration.Builder.class)

public final class PromptToolConfiguration  {
    @Deprecated
    @java.beans.ConstructorProperties({"llm", "promptText", "modelSettings"})
    public PromptToolConfiguration(LlmConfig llm, String promptText, java.util.Map<String, Object> modelSettings) {
        super();
        this.llm = llm;
        this.promptText = promptText;
        this.modelSettings = modelSettings;
    }

    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
        
        @com.fasterxml.jackson.annotation.JsonProperty("llm")
        private LlmConfig llm;

        

        public Builder llm(LlmConfig llm) {
        this.llm = llm;
        return this;
        }
            /**
     * The generative AI prompt
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("promptText")
        private String promptText;

                /**
         * The generative AI prompt
         * @param promptText the value to set
         * @return this builder
         **/
        

        public Builder promptText(String promptText) {
        this.promptText = promptText;
        return this;
        }
            /**
     * Model specific inference parameters such as temperature, top-k, max length, response format, etc.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("modelSettings")
        private java.util.Map<String, Object> modelSettings;

                /**
         * Model specific inference parameters such as temperature, top-k, max length, response format, etc.
         * @param modelSettings the value to set
         * @return this builder
         **/
        

        public Builder modelSettings(java.util.Map<String, Object> modelSettings) {
        this.modelSettings = modelSettings;
        return this;
        }


        public PromptToolConfiguration build() {
            PromptToolConfiguration model = new PromptToolConfiguration(this.llm
                    , this.promptText
                    , this.modelSettings);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(PromptToolConfiguration model) {
                this.llm(model.getLlm());
    this.promptText(model.getPromptText());
    this.modelSettings(model.getModelSettings());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    


    
    @com.fasterxml.jackson.annotation.JsonProperty("llm")
    private final LlmConfig llm;

    
    public LlmConfig getLlm() {
        return llm;
    }


        /**
     * The generative AI prompt
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("promptText")
    private final String promptText;

        /**
     * The generative AI prompt
     * @return the value
     **/
    
    public String getPromptText() {
        return promptText;
    }


        /**
     * Model specific inference parameters such as temperature, top-k, max length, response format, etc.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("modelSettings")
    private final java.util.Map<String, Object> modelSettings;

        /**
     * Model specific inference parameters such as temperature, top-k, max length, response format, etc.
     * @return the value
     **/
    
    public java.util.Map<String, Object> getModelSettings() {
        return modelSettings;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("PromptToolConfiguration(");
        sb.append("llm=").append(String.valueOf(this.llm));
        sb.append(", promptText=").append(String.valueOf(this.promptText));
        sb.append(", modelSettings=").append(String.valueOf(this.modelSettings));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof PromptToolConfiguration)) {
            return false;
        }

        PromptToolConfiguration other = (PromptToolConfiguration) o;
        return java.util.Objects.equals(this.llm, other.llm) &&
            java.util.Objects.equals(this.promptText, other.promptText) &&
            java.util.Objects.equals(this.modelSettings, other.modelSettings);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        result = (result * PRIME) + (this.llm == null ? 43 : this.llm.hashCode());
        result = (result * PRIME) + (this.promptText == null ? 43 : this.promptText.hashCode());
        result = (result * PRIME) + (this.modelSettings == null ? 43 : this.modelSettings.hashCode());
        return result;
    }


}

File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/prompt_tool_configuration.py
# coding: utf-8
# Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.



from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class PromptToolConfiguration(object):
    """
    Tool configurations are set by the agent developer when they create the tool. | The agent does not see those configurations and can not modify their values
    """

    def __init__(self, **kwargs):
        """
        Initializes a new PromptToolConfiguration object with values from keyword arguments.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param llm:
            The value to assign to the llm property of this PromptToolConfiguration.
        :type llm: oci.aidataplatform_dp.models.LlmConfig

        :param prompt_text:
            The value to assign to the prompt_text property of this PromptToolConfiguration.
        :type prompt_text: str

        :param model_settings:
            The value to assign to the model_settings property of this PromptToolConfiguration.
        :type model_settings: dict(str, object)

        """
        self.swagger_types = {
            'llm': 'LlmConfig',
            'prompt_text': 'str',
            'model_settings': 'dict(str, object)'
        }

        self.attribute_map = {
            'llm': 'llm',
            'prompt_text': 'promptText',
            'model_settings': 'modelSettings'
        }

        self._llm = None
        self._prompt_text = None
        self._model_settings = None

    @property
    def llm(self):
        """
        Gets the llm of this PromptToolConfiguration.

        :return: The llm of this PromptToolConfiguration.
        :rtype: oci.aidataplatform_dp.models.LlmConfig
        """
        return self._llm

    @llm.setter
    def llm(self, llm):
        """
        Sets the llm of this PromptToolConfiguration.

        :param llm: The llm of this PromptToolConfiguration.
        :type: oci.aidataplatform_dp.models.LlmConfig
        """
        self._llm = llm

    @property
    def prompt_text(self):
        """
        Gets the prompt_text of this PromptToolConfiguration.
        The generative AI prompt


        :return: The prompt_text of this PromptToolConfiguration.
        :rtype: str
        """
        return self._prompt_text

    @prompt_text.setter
    def prompt_text(self, prompt_text):
        """
        Sets the prompt_text of this PromptToolConfiguration.
        The generative AI prompt


        :param prompt_text: The prompt_text of this PromptToolConfiguration.
        :type: str
        """
        self._prompt_text = prompt_text

    @property
    def model_settings(self):
        """
        Gets the model_settings of this PromptToolConfiguration.
        Model specific inference parameters such as temperature, top-k, max length, response format, etc.


        :return: The model_settings of this PromptToolConfiguration.
        :rtype: dict(str, object)
        """
        return self._model_settings

    @model_settings.setter
    def model_settings(self, model_settings):
        """
        Sets the model_settings of this PromptToolConfiguration.
        Model specific inference parameters such as temperature, top-k, max length, response format, etc.


        :param model_settings: The model_settings of this PromptToolConfiguration.
        :type: dict(str, object)
        """
        self._model_settings = model_settings

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other
